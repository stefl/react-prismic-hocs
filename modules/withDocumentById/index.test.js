import React from 'react'
import withDocumentById from './'
import renderer from 'react-test-renderer'
import { XMLHttpRequest } from 'xmlhttprequest'
import Receiver from '../test/receiver'

window.XMLHttpRequest = XMLHttpRequest

const url = 'https://lesbonneschoses-ve9ubyaaaciapezy.prismic.io/api/v2'
const id = 'UlfoxUnM0wkXYXbU' 

describe ('The query component', () => {
  test('When passed url', (done) => {
    const Query = withDocumentById({
      url,
      id
    })(Receiver)
    renderer.create(<Query done={done} />)
  })
  test('When passed url and query key', (done) => {
    const Query = withDocumentById({
      url,
      id,
      queryKey: 'test'
    })(Receiver)
    renderer.create(<Query done={done} />)
  })
  test('Throws an invariant with no url prop', (done) => {
    const Query = withDocumentById({
    })(Receiver)

    try {
      renderer.create(<Query done={()=>null} />)
      throw new Error()
    } catch (e) {
      expect(e.name).toMatch(/Invariant/)
      done()
    }
  })
  test('Throws an invariant with id', (done) => {
    const Query = withDocumentById({
      url
    })(Receiver)
    try {
      renderer.create(<Query done={()=>null} />)
      throw new Error()
    } catch (e) {
      expect(e.name).toMatch(/Invariant/)
      done()
    }
  })
})

