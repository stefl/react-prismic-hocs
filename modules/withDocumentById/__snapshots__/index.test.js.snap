// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The query component Throws an invariant with id 1`] = `false`;

exports[`The query component Throws an invariant with no url prop 1`] = `false`;

exports[`The query component When passed url 1`] = `false`;

exports[`The query component When passed url 2`] = `
Object {
  "alternate_languages": Array [],
  "data": Object {
    "store": Object {
      "address": Object {
        "type": "Text",
        "value": "2-29-1 Dougenzaka",
      },
      "address2": Object {
        "type": "Text",
        "value": "Shibuya-ku",
      },
      "city": Object {
        "type": "Text",
        "value": "Tokyo",
      },
      "country": Object {
        "type": "Select",
        "value": "Japan",
      },
      "description": Object {
        "type": "StructuredText",
        "value": Array [
          Object {
            "spans": Array [],
            "text": "Your traditional pastries on the go, in the beating heart of the rising sun's city life.",
            "type": "paragraph",
          },
        ],
      },
      "friday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "image": Object {
        "type": "Image",
        "value": Object {
          "main": Object {
            "alt": null,
            "copyright": null,
            "dimensions": Object {
              "height": 500,
              "width": 1500,
            },
            "url": "https://lesbonneschoses-ve9ubyaaaciapezy.cdn.prismic.io/lesbonneschoses/babc76721887f8e623061f9e80c43edb1d6e2d40.jpg",
          },
          "views": Object {
            "icon": Object {
              "alt": null,
              "copyright": null,
              "dimensions": Object {
                "height": 250,
                "width": 250,
              },
              "url": "https://lesbonneschoses-ve9ubyaaaciapezy.cdn.prismic.io/lesbonneschoses/bb48218082621ed3d46e58764fce1750c61b4045.jpg",
            },
            "medium": Object {
              "alt": null,
              "copyright": null,
              "dimensions": Object {
                "height": 250,
                "width": 800,
              },
              "url": "https://lesbonneschoses-ve9ubyaaaciapezy.cdn.prismic.io/lesbonneschoses/c10fcfc9a884e32ecbb3e7dbc9f7cf6157b28069.jpg",
            },
          },
        },
      },
      "monday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "name": Object {
        "type": "StructuredText",
        "value": Array [
          Object {
            "spans": Array [],
            "text": "Tokyo Roppongi Hills",
            "type": "heading1",
          },
        ],
      },
      "saturday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "sunday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "thursday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "tuesday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "wednesday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "zipcode": Object {
        "type": "Text",
        "value": "150-0043",
      },
    },
  },
  "first_publication_date": null,
  "href": "https://lesbonneschoses-ve9ubyaaaciapezy.cdn.prismic.io/api/v1/documents/search?ref=WHzuVR8AAL5VPhhf&q=%5B%5B%3Ad+%3D+at%28document.id%2C+%22UlfoxUnM0wkXYXbU%22%29+%5D%5D",
  "id": "UlfoxUnM0wkXYXbU",
  "lang": "en-us",
  "last_publication_date": null,
  "linked_documents": Array [],
  "slugs": Array [
    "tokyo-roppongi-hills",
  ],
  "tags": Array [],
  "type": "store",
  "uid": null,
}
`;

exports[`The query component When passed url and query key 1`] = `
Object {
  "alternate_languages": Array [],
  "data": Object {
    "store": Object {
      "address": Object {
        "type": "Text",
        "value": "2-29-1 Dougenzaka",
      },
      "address2": Object {
        "type": "Text",
        "value": "Shibuya-ku",
      },
      "city": Object {
        "type": "Text",
        "value": "Tokyo",
      },
      "country": Object {
        "type": "Select",
        "value": "Japan",
      },
      "description": Object {
        "type": "StructuredText",
        "value": Array [
          Object {
            "spans": Array [],
            "text": "Your traditional pastries on the go, in the beating heart of the rising sun's city life.",
            "type": "paragraph",
          },
        ],
      },
      "friday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "image": Object {
        "type": "Image",
        "value": Object {
          "main": Object {
            "alt": null,
            "copyright": null,
            "dimensions": Object {
              "height": 500,
              "width": 1500,
            },
            "url": "https://lesbonneschoses-ve9ubyaaaciapezy.cdn.prismic.io/lesbonneschoses/babc76721887f8e623061f9e80c43edb1d6e2d40.jpg",
          },
          "views": Object {
            "icon": Object {
              "alt": null,
              "copyright": null,
              "dimensions": Object {
                "height": 250,
                "width": 250,
              },
              "url": "https://lesbonneschoses-ve9ubyaaaciapezy.cdn.prismic.io/lesbonneschoses/bb48218082621ed3d46e58764fce1750c61b4045.jpg",
            },
            "medium": Object {
              "alt": null,
              "copyright": null,
              "dimensions": Object {
                "height": 250,
                "width": 800,
              },
              "url": "https://lesbonneschoses-ve9ubyaaaciapezy.cdn.prismic.io/lesbonneschoses/c10fcfc9a884e32ecbb3e7dbc9f7cf6157b28069.jpg",
            },
          },
        },
      },
      "monday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "name": Object {
        "type": "StructuredText",
        "value": Array [
          Object {
            "spans": Array [],
            "text": "Tokyo Roppongi Hills",
            "type": "heading1",
          },
        ],
      },
      "saturday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "sunday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "thursday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "tuesday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "wednesday": Array [
        Object {
          "type": "Text",
          "value": "9am - 10pm",
        },
      ],
      "zipcode": Object {
        "type": "Text",
        "value": "150-0043",
      },
    },
  },
  "first_publication_date": null,
  "href": "https://lesbonneschoses-ve9ubyaaaciapezy.cdn.prismic.io/api/v1/documents/search?ref=WHzuVR8AAL5VPhhf&q=%5B%5B%3Ad+%3D+at%28document.id%2C+%22UlfoxUnM0wkXYXbU%22%29+%5D%5D",
  "id": "UlfoxUnM0wkXYXbU",
  "lang": "en-us",
  "last_publication_date": null,
  "linked_documents": Array [],
  "slugs": Array [
    "tokyo-roppongi-hills",
  ],
  "tags": Array [],
  "type": "store",
  "uid": null,
}
`;
